x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -101,6 +101,13 @@
 import static org.elasticsearch.xpack.esql.plan.logical.join.InlineJoin.firstSubPlan;
 import static org.elasticsearch.xpack.esql.session.SessionUtils.checkPagesBelowSize;
 
+/**
+ * Combines all components necessary for the coordinating node to plan and execute an ESQL query.
+ * <p>
+ * Note that this is not a session in the traditional sense of a stateful connection. This will
```

Comment:
`Note that this is not a session`, I wonder if it  justifies to rename the class to something else.